src/field/field_factory.cxx

```diff
@@ -97,7 +97,13 @@ FieldFactory::~FieldFactory() {
 }
 
 const Field2D FieldFactory::create2D(const string &value, Options *opt, Mesh *m, CELL_LOC loc, BoutReal t) {
-  Field2D result = 0.;
+
+  if(m == NULL)
+    m = fieldmesh;
+  if(m == NULL)
+    throw BoutException("Not a valid mesh");
+
+  Field2D result(0.,m);
 
   if(mesh->StaggerGrids == false){
```

Comment:
We should probably be consistent and use the passed in `Mesh` everywhere in this function, rather than the global `mesh`